src/packet/buffer_tx.rs

```diff
@@ -165,6 +203,80 @@ impl PacketizingBuffer {
     }
 }
 
+// Total queue time in buffer. This lovely drawing explains how to add more time.
+//
+// -time--------------------------------------------------------->
+//
+// +--------------+
+// |              |
+// +--------------+
+//      +---------+                          Already
+//      |         |                           queued
+//      +---------+                         durations
+//          +-----+
+//          |     |
+//          +-----+
+//                       +-+
+//                       | |         <-----  Add next
+//                       +-+                  packet
+//
+//
+//
+// +--------------+--------+
+// |              |@@@@@@@@|
+// +--------------+--------+
+//      +---------+--------+                 The @ is
+//      |         |@@@@@@@@|                  what's
+//      +---------+--------+                  added
+//          +-----+--------+
+//          |     |@@@@@@@@|
+//          +-----+--------+
+//                       +-+
+//                       |@|
+//                       +-+
+#[derive(Debug, Default)]
+struct TotalQueue {
```

Comment:
Maybe some doc comments on the fields here

Reply:
sure